lua/telescope/builtin/git.lua

```diff
@@ -98,26 +92,42 @@ git.branches = function(opts)
     upstream = 0,
     committerdate = 0,
   }
-  local register_entry = function(entry, trim_refname_prefix)
-    entry.name = string.sub(entry.refname, string.len(trim_refname_prefix)+1)
+  local unescape_single_quote = function(v)
+      return string.gsub(v, "\\([\\'])", "%1")
+  end
+  local parse_line = function(line)
+    local fields = vim.split(string.sub(line, 2, -2), "''", true)
+    local entry = {
+      head = fields[1],
+      refname = unescape_single_quote(fields[2]),
+      authorname = unescape_single_quote(fields[3]),
+      upstream = unescape_single_quote(fields[4]),
+      committerdate = fields[5],
+    }
+    local prefix
+    if vim.startswith(entry.refname, 'refs/remotes/') then
+      prefix = 'refs/remotes/'
+    elseif vim.startswith(entry.refname, 'refs/heads/') then
+      prefix = 'refs/heads/'
+    else
+      return
+    end
+    local index = 1
+    if entry.head ~= '*' then
+      index = #results + 1
+    end
+
+    entry.name = string.sub(entry.refname, string.len(prefix)+1)
     for key, value in pairs(widths) do
       widths[key] = math.max(value, vim.fn.strdisplaywidth(entry[key]))
```

Comment:
we also have `strdisplaywidth` in `utils.strdisplaywidth` which uses `ffi` instead of `vim.fn.*`. Would be nice if you can change this as well :)

Reply:
fixed.